application/lib/data_model.dart

```diff
@@ -309,18 +309,16 @@ class LandmarksPropertyValue extends PropertyValue {
   @override
   void registerChildren(Atom parent) {
     for (final Landmark landmark in value.where(Landmark.hasChild))
-      landmark.atom.registerFriend(parent);
+      landmark.atom!.registerFriend(parent);
   }
 
   @override
   void unregisterChildren(Atom parent) {
     for (final Landmark landmark in value.where(Landmark.hasChild))
-      landmark.atom.unregisterFriend(parent);
+      landmark.atom!.unregisterFriend(parent);
```

Comment:
how do we know the atom is non-null here?

Reply:
landmark.hasChild guarantees it